Tests/ApolloTests/ApolloClientOperationTests.swift

```diff
@@ -41,21 +41,59 @@ final class ApolloClientOperationTests: XCTestCase {
     }
   }
 
-  func test__performMutation_givenPublishResultToStore_false_doesNotPublishResultsToStore() throws {
-    // given
-    class GivenSelectionSet: MockSelectionSet {
+  // given
+  class GivenSelectionSet: MockSelectionSet {
+    override class var __selections: [Selection] { [
+      .field("createReview", CreateReview.self)
+    ] }
+
+    class CreateReview: MockSelectionSet {
       override class var __selections: [Selection] { [
-        .field("createReview", CreateReview.self)
+        .field("__typename", String.self),
+        .field("stars", Int.self),
+        .field("commentary", String?.self)
       ] }
+    }
+  }
 
-      class CreateReview: MockSelectionSet {
-        override class var __selections: [Selection] { [
-          .field("__typename", String.self),
-          .field("stars", Int.self),
-          .field("commentary", String?.self)
-        ] }
-      }
+  func test__performMutation_givenPublishResultToStore_true_publishResultsToStore() throws {
+    let mutation = MockMutation<GivenSelectionSet>()
+    let resultObserver = self.makeResultObserver(for: mutation)
+
+    let serverRequestExpectation = self.server.expect(MockMutation<GivenSelectionSet>.self) { _ in
+      [
+        "data": [
+          "createReview": [
+            "__typename": "Review",
+            "stars": 3,
+            "commentary": ""
+          ]
+        ]
+      ]
     }
+
+    let performResultFromServerExpectation =
+      resultObserver.expectation(description: "Mutation was successful") { result in
+        switch (result) {
+        case .success:
+          break
+        case let .failure(error):
+          fail("Unexpected failure! \(error)")
+        }
+      }
+
+    // when
+    self.client.perform(mutation: mutation,
+                        publishResultToStore: true,
+                        resultHandler: resultObserver.handler)
+
+    self.wait(for: [serverRequestExpectation, performResultFromServerExpectation], timeout: 0.2)
+
+    // then
+    expect(self.store.publishedRecordSets).notTo(beEmpty())
```

Comment:
I think it would be better to test specifically for a single entry into the store and for the record that was added. This ensures that if somehow there is a different record written to the store this test will then fail.

Reply:
Thanks for your suggestion!
fixed: [5b9df5f](https://github.com/apollographql/apollo-ios/pull/2953/commits/5b9df5ff6c1bc1389c7d8a3cb5fa85e16b800708)